herbie/settings.py

```diff
@@ -0,0 +1,205 @@
+"""
+Django settings for herbie project.
+
+Generated by 'django-admin startproject' using Django 2.2.7.
+
+For more information on this file, see
+https://docs.djangoproject.com/en/2.2/topics/settings/
+
+For the full list of settings and their values, see
+https://docs.djangoproject.com/en/2.2/ref/settings/
+"""
+
+import os
+import environ
+
+env = environ.Env()
+
+# Business entity application label
+APP_LABEL = 'herbieapp'
+
+# Json schema package for validation of business objects
+SCHEMA_REGISTRY_PACKAGE = 'schema'
+
+# chunk size for exporting data
+DEFAULT_CHUNK_SIZE = 100
+
+# message_provider
+MESSAGING_PROVIDER = 'google_pubsub'
+
+# Google Cloud Pub/Sub
+GCLOUD_PUBSUB_PROJECT_ID = env.str('GCLOUD_PUBSUB_PROJECT_ID', " ")
+
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/2.2/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = env.str('SECRET_KEY')
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = False
+
+ALLOWED_HOSTS = ['*']
+
+
+INSTALLED_APPS = [
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'herbie_core.apps.HerbieCoreConfig',
+    'herbieapp.apps.HerbieAppConfig',
+    'google_pubsub_adapter.apps.HerbieGooglePubsubAdapterConfig',
+    'django.contrib.admin',
+    'rest_framework',
+    'rest_framework.authtoken'
+]
+# Application definition
+
+
+MIDDLEWARE = [
+    'django.middleware.security.SecurityMiddleware',
+    'django.contrib.sessions.middleware.SessionMiddleware',
+    'django.middleware.common.CommonMiddleware',
+    'django.middleware.csrf.CsrfViewMiddleware',
+    'django.contrib.auth.middleware.AuthenticationMiddleware',
+    'django.contrib.messages.middleware.MessageMiddleware',
+    'django.middleware.clickjacking.XFrameOptionsMiddleware',
+]
+
+ROOT_URLCONF = 'herbie.urls'
+
+
+SOCIAL_AUTH_URL_NAMESPACE = 'social'
+SOCIAL_AUTH_PIPELINE = (
+    'social_core.pipeline.social_auth.social_details',
+    'social_core.pipeline.social_auth.social_uid',
+    'social_core.pipeline.social_auth.auth_allowed',
+    'social_core.pipeline.social_auth.social_user',
+    'social_core.pipeline.user.get_username',
+    'social_core.pipeline.user.create_user',
+    'social_core.pipeline.social_auth.associate_user',
+    'social_core.pipeline.social_auth.load_extra_data',
+    'social_core.pipeline.user.user_details',
+    'herbieapp.authentication.social_authentication.process_roles'
+)
```

Comment:
I am still not sure if we really want to include the social auth login per default for the admin panel. What do you think about it?

Reply:
Yeah, maybe we can have it as an optional feature instead.